src/MIDebugEngine/Engine.Impl/DebuggedThread.cs

```diff
@@ -147,15 +171,35 @@ internal void MarkDirty()
             }
         }
 
-        internal void ThreadCreatedEvent(int id, string groupId)
+        internal async void ThreadCreatedEvent(int id, string groupId)
         {
-            lock (_threadList)
+            ResultValue resVal = null;
```

Comment:
You need a try/catch around this method now, as async void methods can never let exceptions leave